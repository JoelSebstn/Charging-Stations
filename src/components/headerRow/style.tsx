import { StyleSheet } from "react-native";

export const Styles = StyleSheet.create({
    alignCentre: { flex: 1, alignItems: 'center' },
    flexRow: { flexDirection: 'row' },
    textFont22: { fontSize: 22 },
    horizontalLine: {
        borderBottomColor: 'black',
        borderBottomWidth: 2,
    },
});