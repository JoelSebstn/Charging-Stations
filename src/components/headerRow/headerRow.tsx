import { View, Text } from "react-native";
import { Styles } from "./style";

export function HeaderRow() {
    return (
        <>
            <View style={Styles.flexRow}>
                <View style={Styles.alignCentre} testID=""><Text style={Styles.textFont22}>Name</Text></View>
                <View style={Styles.alignCentre}><Text style={Styles.textFont22}> Status</Text></View>
            </View>
            <View
                style={Styles.horizontalLine}
            />
        </>
    )
}