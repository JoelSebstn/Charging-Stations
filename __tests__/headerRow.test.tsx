import { render, screen } from "@testing-library/react-native";
import { HeaderRow } from "../src/components/headerRow/headerRow";
import renderer from 'react-test-renderer';

describe('TEst Header Row', () => {
    it('should render name', () => {
        const { getByText } = render(
            <HeaderRow />,
        );
        const label = getByText("Name");
        expect(label).toBeTruthy();
    });
    it('should render the Status', () => {
        const { getByText } = render(
            <HeaderRow />,
        );
        const label = getByText("Status");
        expect(label).toBeTruthy();
    });
    it('renders the correct message', () => {
        const { queryByText } = render(<HeaderRow />);
        expect(queryByText('Status')).not.toBeNull();
    });
    it("Matches Header Snapshot", () => {
        const domTree = renderer.create(<HeaderRow />).toJSON();
        expect(domTree).toMatchSnapshot();
    });
})