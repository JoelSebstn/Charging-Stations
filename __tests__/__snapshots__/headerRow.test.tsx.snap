// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEst Header Row Matches Header Snapshot 1`] = `
[
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
        }
      }
      testID=""
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
        Name
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
         Status
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 2,
      }
    }
  />,
]
`;

exports[`asd Matches Header Snapshot 1`] = `
[
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
        Name
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 22,
          }
        }
      >
         Status
      </Text>
    </View>
  </View>,
  <View
    style={
      {
        "borderBottomColor": "black",
        "borderBottomWidth": 2,
      }
    }
  />,
]
`;
